server/src/main/java/com/saralein/server/request/parser/RequestParser.java

```diff
@@ -0,0 +1,55 @@
+package com.saralein.server.request.parser;
+
+import com.saralein.server.request.Request;
+import com.saralein.server.request.transfer.RequestLine;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.saralein.server.Constants.CRLF;
+
+public class RequestParser {
+    private final RequestLineParser requestLineParser;
+    private final HeaderParser headerParser;
+
+    public RequestParser(RequestLineParser requestLineParser, HeaderParser headerParser) {
+        this.requestLineParser = requestLineParser;
+        this.headerParser = headerParser;
+    }
+
+    public Request parse(String rawRequest) throws Exception {
+        if (rawRequest.isEmpty()) {
+            throw new Exception("Bad request. Connection closed.");
```

Comment:
🤔 Weird that parser has this domain knowledge about what the servers going to do with this exception.

Reply:
I think I see what you are saying.  You mean, parser knows it is a bad request.  Parser would have no clue that a connection would be closed.  Right?  Same will apply for the request line parser then.

Reply:
Yeah, that's exactly it